geode-common/build.gradle

```diff
@@ -26,6 +26,7 @@ dependencies {
   implementation(platform(project(':boms:geode-all-bom')))
   implementation('com.fasterxml.jackson.core:jackson-databind')
 
+  api('org.jetbrains:annotations')
```

Comment:
https://github.com/JetBrains/java-annotations suggests `compileOnly` rather than `api` ... is the intent to ship jetbrains annotations .jar with Geode, or use it only at compile time?  I could see arguments either way (or maybe for technical reasons we don't actually have a choice?), just want to be sure we get the right scope because scopes are tricky.  @rhoughton-pivot any thoughts?

Reply:
I concur, it should be compileOnly.

Reply:
I want to introduce a compile-time dependency on `annotation`, in any subproject that depends on `geode-common`, because I want to save my colleagues from having to add an explicit dependency on `annotation` to the `build.gradle` for each subproject, as they desire to add a `@NotNull` annotation in some new place.

I tried changing this to `compileOnly('org.jetbrains:annotations')` but now the compiler/IDE says  `java: package org.jetbrains.annotations does not exist` where I'm trying to import it over in `SocketMessageWriter.java` in `geode-core`.

Reply:
@kohlmu-pivotal wanted the same thing: to have a java project or .gradle file, whose only goal was to make it easy for other Geode subprojects to get dependencies. Maybe he found a solution?

Reply:
I'm calling this resolved. I gave up trying to make the compile-time dependency global since it seems kinda harder than it's worth.